Gems/AtomTressFX/Code/Components/HairComponentController.cpp

```diff
@@ -0,0 +1,373 @@
+                required.push_back(AZ_CRC_CE("EMotionFXActorService"));
+            }
+
+            HairComponentController::HairComponentController(const HairComponentConfig& config)
+                : m_configuration(config)
+            {
+            }
+
+
+            void HairComponentController::Activate(EntityId entityId)
+            {
+                m_entityId = entityId;
+
+                m_featureProcessor = RPI::Scene::GetFeatureProcessorForEntity<Hair::HairFeatureProcessor>(m_entityId);
+                if (m_featureProcessor)
+                {
+                    m_featureProcessor->SetHairGlobalSettings(m_configuration.m_hairGlobalSettings);
+                    if (!m_renderObject)
+                    {
+                        // Call this function if object doesn't exist to trigger the load of the existing asset
+                        OnHairAssetChanged();
+                    }
+                }
+
+                EMotionFX::Integration::ActorComponentNotificationBus::Handler::BusConnect(m_entityId);
+                HairRequestsBus::Handler::BusConnect(m_entityId);
+                TickBus::Handler::BusConnect();
+                HairGlobalSettingsNotificationBus::Handler::BusConnect();
+            }
+
+            void HairComponentController::Deactivate()
+            {
+                HairRequestsBus::Handler::BusDisconnect(m_entityId);
+                EMotionFX::Integration::ActorComponentNotificationBus::Handler::BusDisconnect(m_entityId);
+                Data::AssetBus::MultiHandler::BusDisconnect();
+                TickBus::Handler::BusDisconnect();
+                HairGlobalSettingsNotificationBus::Handler::BusDisconnect();
+
+                RemoveHairObject();
+                m_entityId.SetInvalid();
+            }
+
+            void HairComponentController::SetConfiguration(const HairComponentConfig& config)
+            {
+                m_configuration = config;
+                OnHairConfigChanged();
+            }
+
+            const HairComponentConfig& HairComponentController::GetConfiguration() const
+            {
+                return m_configuration;
+            }
+
+            void HairComponentController::OnHairAssetChanged()
+            {
+                Data::AssetBus::MultiHandler::BusDisconnect();
+                if (m_configuration.m_hairAsset.GetId().IsValid())
+                {
+                    Data::AssetBus::MultiHandler::BusConnect(m_configuration.m_hairAsset.GetId());
+                    m_configuration.m_hairAsset.QueueLoad();
+                }
+                else
+                {
+                    RemoveHairObject();
+                }
+            }
+
+            void HairComponentController::OnHairGlobalSettingsChanged(const HairGlobalSettings& hairGlobalSettings)
+            {
+                m_configuration.m_hairGlobalSettings = hairGlobalSettings;
+            }
```

Comment:
Very good question.
All of this is indirectly triggered by the editor property change that will have callback to HairComponentConfig::OnHairGlobalSettingsChanged.
This will start a chain of events that will eventually trigger the FeatureProcessor that will fire another callback to update global settings for all hair components - this is the one downside of having it not as a single global settings component.
For a single object what this chain of event does is basically set the property again, but for multiple hair objects it'll make sure to set all components with the same general properties.